### ISSUE TITLE
Archive file contains unreadable content

### ISSUE THREAD

**Issue by bwstaal** [NLog/NLog#922]
Just as the title says: the log file that gets created by NLog as part of the archiving process on the startup of the application contains garbage and is therefore unuable.

By 'garbage' I mean:
```
PK    Ðr5GC½Bót   {      NLog_Vfpf_Technical.txt-Ê½‚0 àÝÄwèp¡Å_VHÜ¸â¼’»—>¼‹ßü¹Ú«úZ9kì¡u®m8Ùó¥t{_ža
Ðq¯Iò˜² (¨0PVèyBùÒ[‰5á²ÏÿRn‘7”d>˜^qBã=Ï(WdØï~PK     Ðr5GC½Bót   {                    NLog_Vfpf_Technical.txtPK      E   ©
```

I'm using the latest version 4.1.2 with the following config:

```
 <nlog throwExceptions="true"
        internalLogFile="C:\LogFiles\NLog_Internal.txt"
        internalLogLevel="Trace"
        autoReload="true">
    <targets>
      <!-- 5012000 = 5MB -->
      <target name="technicalLog"
              type="File" 
              layout="${longdate}|${level:uppercase=true}|${logger}|${message}"
              fileName="C:\LogFiles\NLog_Technical.txt"
              archiveFileName="C:\LogFiles\NLog_Technical_{####}.txt"
              archiveAboveSize="5"
              archiveNumbering="Date"
              maxArchiveFiles="7" 
              keepFileOpen="true"
              enableArchiveFileCompression="true" />
    </targets>
    <rules>
      <logger name="*" minlevel="Trace" writeTo="technicalLog" />
    </rules>
  </nlog>
```

**Comment by 304NotModified** [comment]
It's creating a `.Zip` file.  Did you tried opening with a zip application, eg 7zip?

**Comment by bwstaal** [comment]
No I didn't. But after changing the extension from ```.Txt``` to ```.Zip``` I saw that it has created a usable zipfile. I looked through the code for the archive file compression functionality and I saw that I had to specify the ```.Zip``` extenstion in the ```archiveFileName``` attribute in config, otherwise it's creating the file ```.Zip``` file with the wrong extension.

**Comment by 304NotModified** [comment]
> . I looked through the code for the archive file compression functionality and I saw that I had to specify the .Zip extenstion in the archiveFileName attribute in config

I don't fully understand this. I you name it file.txt and rename it to file.zip, then I should work?

**Comment by bwstaal** [comment]
Yes, then it works.

**Comment by 304NotModified** [comment]
Ps. We fixed this in NLog 4.2. The zip is added to the archive if you don't specify a name for the archive file.

**Comment by bwstaal** [comment]
I saw that you fixed it. Thanks!
